__std__/flake.nix

```diff
@@ -0,0 +1,248 @@
+    sphinxcontrib-haddock = {
+      url = "github:michaelpj/sphinxcontrib-haddock";
+      flake = false;
+    };
+    # cardano-repo-tool = {
+    #   url = "github:input-output-hk/cardano-repo-tool";
+    #   flake = false;
+    # };
+    # gitignore-nix = {
+    #   url = "github:hercules-ci/gitignore.nix";
+    #   flake = false;
+    # };
+    # haskell-language-server = {
+    #   # Pinned to a release
+    #   url = "github:haskell/haskell-language-server?ref=1.7.0.0";
+    #   flake = false;
+    # };
+    # iohk-nix = {
+    #   url = "github:input-output-hk/iohk-nix";
+    #   flake = false;
+    # };
+    # pre-commit-hooks-nix = {
+    #   url = "github:cachix/pre-commit-hooks.nix";
+    #   flake = false;
+    # };
+  };
+
+  # The flake outputs are managed by std.
+  outputs = inputs:
+
+    # The growOn function accepts a first "soil" argument, defining the organelles.
+    inputs.std.growOn
+      {
+
+        # Boilerplate
+        inherit inputs;
+
+        # ALL nix files will reside inside this folder, no exception.
+        # Each subfolder is a "cell".
+        # Cell names are arbitrary.
+        # Cells are for highest-level organization and grouping of nix code.
+        #
+        # In this repository we have three cells:
+        #   doc
+        #     Develop and build all the documentation artifacts
+        #   haskell
+        #     Develop and build all haskell components
+        #   toolchain
+        #     Common tools and functions shared across multiple cells
+        cellsFrom = ./nix;
+
+        # Each cell contains arbitrary "organelles".
+        # Each organelle must be either:
+        #   A nix file named after the organelle
+        #   A directory named after the organelle and containing a default.nix
+        # Organelles have types.
+        # Not all cells have the same organelles.
+        # All organelles belong in a cell.
+        #
+        # In this repository we have five organelles, listed below with their type:
+        #   devshells :: devshells
+        #     Development shells available via nix develop
+        #   packages :: installables
+        #     Packages available via nix build
+        #   devshellProfiles :: functions
+        #     Building blocks for devshells, not exposed by the flake
+        #   scripts :: functions
+        #     Bash scripts simplifying or automating a variety of tasks
+        #     Generally these are available as commands inside the development shell
+        #     These are very repository specific, and are not exposed by the flake
+        #   library :: functions
+        #     Functions and values shared across the current cell
+        #     These are very repository specific, and are not exposed by the flake
+        #
+        # std provides a TUI to interact with the organelles.
+        # Available interactions are determined by the organelle's type.
+        # Because this repository does not use the TUI, the type is actually irrelevant.
```

Comment:
irrelevant: not entirely.

Ani CI/CD preset (in planning) will autodetect stuff based on these types
Also some Clade actions are relatively useful. Currently most useful: the container clade actions, but that action library will only increase over time.

But for simplicity, this could be omitted or only touched here.